src/niscope/metadata/functions_addon.py

```diff
@@ -186,16 +194,301 @@
+first. For example, with a channel list of 0,1, you would have the
+following index values:
+
+index 0 = record 0, channel 0
+
+index *x* = record 0, channel 1
+
+index 2\ *x* = record 1, channel 0
+
+index 3\ *x* = record 1, channel 1
+
+Where *x* = the record length
+
+Types supported are
+
+- `numpy.float64`
+- `numpy.int8`
+- `numpy.in16`
+- `numpy.int32`
+
+Example:
+
+.. code-block:: python
+
+    wfm = numpy.ndarray(num_samples * session.actual_num_wfms(), dtype=numpy.float64)
+    wfm_info = session['0,1'].fetch_into(num_samples, wfms, timeout=5.0)
+
+''',
+                },
+            },
+            {
+                'direction': 'out',
+                'enum': None,
+                'name': 'wfmInfo',
+                'type': 'struct niScope_wfmInfo[]',
+                'documentation': {
+                    'description': '''
+                    Returns an array of classed with the following timing and scaling information about each waveform:
+
+                    -  **relative_initial_x** the time (in seconds) from the trigger to the first sample in the fetched waveform
+                    -  **absolute_initial_x** timestamp (in seconds) of the first fetched sample. This timestamp is comparable between records and acquisitions; devices that do not support this parameter use 0 for this output.
+                    -  **x_increment** the time between points in the acquired waveform in seconds -  **actual_samples** the actual number of samples fetched and placed in the waveform array
+                    -  **gain** the gain factor of the given channel; useful for scaling binary data with the following formula:
+
+                        .. math::
+
+                            voltage = binary data * gain factor + offset
+
+                    -  **offset** the offset factor of the given channel; useful for scaling binary data with the following formula:
+
+                        .. math::
+
+                            voltage = binary data * gain factor + offset
+
+                    Call niScope\_ActualNumWfms to determine the size of this array.
+''',
+                },
+            },
         ],
+        'documentation': {
+            'description': '''
+Returns the waveform from a previously initiated acquisition that the
+digitizer acquires for the specified channel. This function returns
+scaled voltage waveforms.
+
+This function may return multiple waveforms depending on the number of
+channels, the acquisition type, and the number of records you specify.
+''',
+            'note': '''
+Some functionality, such as time stamping, is not supported in all
+digitizers. Refer to `Features Supported by
+Device <REPLACE_DRIVER_SPECIFIC_URL_1(features_supported_main)>`__ for
+more information.
+''',
+        },
     },
 }
 
+# Override the 'python' name for some functions.
+functions_python_name = {
+    'Fetch':                      { 'python_name': '_fetch_double',                   },
```

Comment:
In nifgen I kept the python names the same when private. Can you keep this as _fetch?

Reply:
That means we would have `fetch`, `fetch_into`, `_fetch` and `_fetch_into`. The first two are hand maintained but different templates and the second two come from Fetch.

Is that okay?

Reply:
I think it's fine.

Reply:
Updated